### PR TITLE
Tganley/bugfix/sband hk no float

### PR DESCRIPTION
**Pull request by tganley**

### CODE REVIEW

File: ex2_hal/sband/equipment_handler/include/sTransmitter.h [hunk 1]
```diff
@@ -91,10 +91,11 @@
 #define S_PAPWR_30DBM 30
 
 // Frequency maximums/minimums
-#define S_FREQ_COMMERCIAL_MAX 2300.0f
-#define S_FREQ_COMMERCIAL_MIN 2200.0f
-#define S_FREQ_AMATEUR_MAX 2450.0f
-#define S_FREQ_AMATEUR_MIN 2400.0f
+#define S_FREQ_COMMERCIAL_MAX 2300000000 // Hz
```
Comment (rcunrau) [inline ex2_hal/sband/equipment_handler/include/sTransmitter.h:94]: This looks like 2.3 Giga Hertz - just confirming that's what you meant it to be :-)

File: ex2_hal/sband/equipment_handler/include/sTransmitter.h [hunk 2]
```diff
@@ -108,12 +108,12 @@
 // Conversion factors
 #define S_FREQ_OFFSET_SCALING 2
 #define S_FWVER_MAJORNUM_SCALING 100
-#define S_OUTPWR_SCALING 0.00114f
-#define S_PATEMP_SCALING 0.07324f
-#define S_PATEMP_OFFSET -50.0f
-#define S_TEMP_SCALING 0.0625f
-#define S_CURRENT_SCALING 0.00004f
-#define S_VOLTAGE_SCALING 0.004F
+#define S_OUTPWR_SCALING 114 / 100000
```
Comment (rcunrau) [inline ex2_hal/sband/equipment_handler/include/sTransmitter.h:111]: Pretty sure these will all be 0. For example 4/5=0 in integer math

**Review by rcunrau — CHANGES_REQUESTED** [review]

File: ex2_hal/sband/equipment_handler/include/sTransmitter.h (hunk 1 above)
Comment (tganley) [reply ex2_hal/sband/equipment_handler/include/sTransmitter.h:94]: Yes it is!

File: ex2_hal/sband/equipment_handler/include/sTransmitter.h (hunk 2 above)
Comment (tganley) [reply ex2_hal/sband/equipment_handler/include/sTransmitter.h:111]: These are used in very specific places and I made of the order of operations. Often a multiplication will happen on the first number, then the division

Comment (Taylorr82) [reply ex2_hal/sband/equipment_handler/include/sTransmitter.h:111]: This isn't really the standard way of doing it. It should be like this
```suggestion
#define S_OUTPWR_SCALING 114 / 100000
```#define S_OUTPWR_SCALING(n) ((n * 114) / 100000)

Comment (tganley) [reply ex2_hal/sband/equipment_handler/include/sTransmitter.h:111]: Ah okay I'll fix that

**Review by Taylorr82 — APPROVED** [review]

**Pull request merged**